Assets/MixedRealityToolkit/Inspectors/Utilities/MixedRealitySearchUtility.cs

```diff
@@ -0,0 +1,225 @@
+        /// </summary>
+        private static readonly HashSet<string> serializedPropertiesToIgnore = new HashSet<string>()
+        {
+            // Unity base class fields
+            "m_Name",
+            "m_Script",
+            "m_Enabled",
+            "m_GameObject",
+            "m_ObjectHideFlags",
+            "m_CorrespondingSourceObject",
+            "m_PrefabInstance",
+            "m_PrefabAsset",
+            "m_EditorHideFlags",
+            "m_EditorClassIdentifier",
+            // Profile base class fields
+            "isCustomProfile",
+        };
+
+        /// <summary>
+        /// Field types that don't need their child properties displayed.
+        /// </summary>
+        private static readonly HashSet<string> serializedPropertyTypesToFlatten = new HashSet<string>()
+        {
+            typeof(SystemType).Name,
+        };
+
+        private const int minSearchStringLength = 3;
+
+        /// <summary>
+        /// Searches all profiles and sub profiles for matching fields. Returns results in unranked order.
+        /// </summary>
+        /// <param name="profile">The configuration profile you're searching.</param>
+        /// <param name="config">Configuration for search.</param>
+        public static IEnumerable<ProfileSearchResult> SearchProfileFields(UnityEngine.Object profile, SearchConfig config)
+        {
+            if (string.IsNullOrEmpty(config.SearchFieldString))
+            {   // If the config is empty, bail early
+                yield break;
+            }
+
+            // Generate keywords if we haven't yet
+            if (config.Keywords == null)
+            {
+                config.Keywords = new HashSet<string>(config.SearchFieldString.Split(new string[] { " ", "," }, StringSplitOptions.RemoveEmptyEntries));
+                config.Keywords.RemoveWhere(s => s.Length < minSearchStringLength);
+            }
+
+            // The result that we will return, if not empty
+            ProfileSearchResult result = new ProfileSearchResult();
+
+            // Ignore these requirements if criteria is empty
+            config.RequireAllKeywords &= config.Keywords.Count > 0;
+
+            // Go through the profile's serialized fields
+            SerializedProperty iterator = new SerializedObject(profile).GetIterator();
+            bool hasNextProperty = iterator.Next(true);
+
+            while (hasNextProperty)
+            {
+                if (serializedPropertiesToIgnore.Contains(iterator.name))
+                {   // We don't care about this property
+                    hasNextProperty = iterator.Next(false);
+                    continue;
+                }
+
+                // Check to see if this is a profile - if it is, enter the profile and check its children
+                bool isProfileField = false;
+                if (iterator.propertyType == SerializedPropertyType.ObjectReference && iterator.objectReferenceValue != null)
```

Comment:
Probably easier to extract this out to IsProfileField() helper function and can combine if statements